backend/src/hatchling/builders/config.py

```diff
@@ -787,9 +787,7 @@ def default_packages(self):
         return []
 
     def default_global_exclude(self):
-        patterns = ['*.py[cdo]', *EXCLUDED_DIRECTORIES]
-        patterns.sort()
-        return patterns
```

Comment:
The directories are excluded during traversal so this pattern matching was wasteful